shared/dev/dumb-sheet.render.native.js

```diff
@@ -40,32 +37,33 @@ class Render extends Component<void, Props, any> {
     }, 0)
   }
 
-  _continueTest () {
-    if (isTesting && this.state.testIndex !== -1) {
+  _increment () {
+    if (this.props.autoIncrement && this.state.testIndex !== -1) {
       const {componentsOnly} = this._getComponents()
       if (this.state.testIndex >= componentsOnly.length) {
         this.setState({testIndex: -1})
       } else {
-        this.setState({testIndex: this.state.testIndex + TEST_RENDER_NUMBER})
+        this.setState({testIndex: this.state.testIndex + 1})
       }
     }
   }
 
   componentDidMount () {
-    if (isTesting) {
-      this._continueTest()
+    if (this.props.autoIncrement) {
+      this._increment()
     }
   }
 
   componentDidUpdate () {
-    setTimeout(() => {
-      this._continueTest()
-    }, 1)
+    if (this.props.autoIncrement) {
+      setTimeout(() => {
+        this._increment()
+      }, 1)
```

Comment:
I need to increment once the render is done so I'd have to plumb that out.
This is just a mode this component is in and this is dev only and never
reused so I don't want to over engineer this thing

On Wednesday, August 31, 2016, Max Goodman <notifications@github.com> wrote:

> In shared/dev/dumb-sheet.render.native.js
> <https://github.com/keybase/client/pull/4053#discussion_r77078747>:
>
> >      }
> >    }
> >
> >    componentDidUpdate () {
> > -    setTimeout(() => {
> > -      this._continueTest()
> > -    }, 1)
> > +    if (this.props.autoIncrement) {
> > +      setTimeout(() => {
> > +        this._increment()
> > +      }, 1)
>
> Ah I see, the advancement is triggered when componentDidUpdate triggers.
>
> That being said, I think the need to setTimeout is a strong smell that
> the sheet increment shouldn't be driven by this function.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/keybase/client/pull/4053/files/dc32ebab67e2b5011be080959d5dc7301771135f..a2530d5f598003cb6416f7a8cff641c1c3ab9514#r77078747>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAfhU0Y0EHOPgjM_mtX8sthpMUASuKQKks5qlfSbgaJpZM4Jtgrl>
> .
>